crates/bevy_ecs/src/storage/resource.rs

```diff
@@ -166,12 +233,14 @@ impl Resources {
         component_id: ComponentId,
         components: &Components,
         f: impl FnOnce() -> ArchetypeComponentId,
-    ) -> &mut ResourceData {
+    ) -> &mut ResourceData<SEND> {
```

Comment:
This function needs a `// Safety` that `component_id` is send if `SEND` is true (or alternatively panic) because we skip the send checks when `SEND=true` but do not have safety invariants (or panics) in every function that the `ComponentId` is `Send` if `SEND` is.